### DISCUSSION TITLE
Make "Url" Card Field allow HTML or open link in new page

### DISCUSSION THREAD

**Discussion by PFederwitz** [microsoft-search/pnp-modern-search#2866] [Ideas]
**Is your feature request related to a problem? Please describe.**
I am using PNP-Search to create a search page for our photo library. I am using the Card layout, but need a better way to link to the photo than the default. When you click on the Title it takes you directly to the photo file. I need it to open a new tab or page because when people view the photo and then click the back button they end up with a new search rather than where they were. 

**Describe the solution you'd like**
The Url link should either allow HTML so that I can add the necessary HTML to make it work ('target="_blank" data-interception="off"') or have an option of some sort that allows opening the link in a new tab. Note that the List layout does this by default, but the card layout does not even allow it as an option.

I previously posted an error with my work around in this issue: https://github.com/microsoft-search/pnp-modern-search/issues/1083

**Describe alternatives you've considered**
I am trying to reuse the Location field to make this work since it allows HTML, but that will be confusing to users. I am changing the link text to "file details" and then sending them to a file properties box in a new window.

**Additional context**
I considered trying to recreate the card layout from scratch so that I could implement this the way that I wanted to, but since it uses the Pnp-documentcard template I can't see the HTML below the surface to recreate my own.

**Comment by wobba** [comment]
I assume this is v4 as in v3 clicking a card open in a new tab. I think we should add this to v4 as well - possible with an option to decide the behavior.

**Comment by PFederwitz** [comment]
Yes, version 4.2.2

**Comment by Ofer-Gal** [comment]
I have version 4.3.0 but do not see the option for target="_blank". What am I missing?

**Comment by wobba** [comment]
@Ofer-Gal you are not missing anything as it's not implemented yet in v4.

**Comment by Ofer-Gal** [comment]
What should I do if I would like to add the option to set the items per page (in all layouts) to be set by the user? now it is only in the web part properties 

Thanks

**Comment by wobba** [comment]
@Ofer-Gal that needs a development change of the web parts. You can suggest this as a new option in a separate issue.

**Comment by baltasaurusrex** [comment]
My workaround is to:

1. wrap the code with an anchor <a> element; (which messed up the styling)
2. so I gave the anchor element the same css class as the pnp-documentcard element (which is "template--card"). 

So I wrap the <pnp-documentcard /> like this:

` <a href="{{slot item @root.slots.PreviewUrl}}" target="_blank" class="template--card" data-interception="off" rel="noopener noreferrer">
   <pnp-documentcard />
</a>`

The attributes for the <a> anchor element are taken from the attributes of the <a> anchor for the Title in the List layout

**Reply by gretchunkim** [reply]
@baltasaurusrex This  is indeed my workaround for most of image/photo issue in Adaptive card views. Good to see it here!
